test/test_helper.rb

```diff
@@ -42,7 +42,7 @@ def assert_template_result_matches(expected, template, assigns = {}, message = n
       assert_match expected, Template.parse(template).render!(assigns), message
     end
 
-    def assert_match_syntax_error(match, template, registers = {})
+    def assert_match_syntax_error(match, template, assigns = {})
       exception = assert_raises(Liquid::SyntaxError) {
         Template.parse(template).render(assigns)
```

Comment:
It worked because `Template.parse` will raise an exception before `render` is called (and its arguments are evaluated).